pkg/reconciler/common/proxy_test.go

```diff
@@ -0,0 +1,131 @@
+/*
+Copyright 2020 The Tekton Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package common
+
+import (
+	"encoding/json"
+	"sort"
+	"testing"
+
+	"gotest.tools/v3/assert"
+	"gotest.tools/v3/env"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+func TestApplyProxySettingsNoProxy(t *testing.T) {
+	actual := unstructuredDeployment(t)
+	expected := unstructuredDeployment(t)
+
+	if err := ApplyProxySettings(actual); err != nil {
+		t.Fatal(err)
+	}
+
+	assert.DeepEqual(t, actual, expected)
+}
+
+func TestApplyProxySettingsWithProxy(t *testing.T) {
+	proxyEnv := map[string]string{
+		"HTTP_PROXY":  "http://1.2.3.4:30001",
+		"HTTPS_PROXY": "http://1.2.3.4:30002",
+		"NO_PROXY":    "index.docker.io",
+	}
+	actual := unstructuredDeployment(t)
+	expected := unstructuredDeployment(t, withEnv(proxyEnv))
+
+	defer env.PatchAll(t, proxyEnv)()
+	if err := ApplyProxySettings(actual); err != nil {
+		t.Fatal(err)
+	}
+
+	assert.DeepEqual(t, actual, expected)
```

Comment:
Bit confused with testcase here
As we have not set env in `actual` how are we getting  `actual` and `expected` as equal ?

Reply:
Because `ApplyProxySettings` does its job 😉 — aka ready the proxy env variable (set by `env.PatchAll`) and add them to the list of environment variables.